Source/UI/IdentityUI.swift

```diff
@@ -477,6 +488,105 @@ extension IdentityUI {
         }
     }
 
+    @available(iOS 13.0, *)
+    private func showSharedWebCredentialsController(client: Client,
+                                                    presentingViewController: UIViewController,
+                                                    localizedTeaserText: String?,
+                                                    scopes: [String],
+                                                    completion: @escaping (Output) -> Void) {
+        func fallback() {
+            show(input: .byLoginMethod(.password,
+                                       presentingViewController: presentingViewController,
+                                       localizedTeaserText: localizedTeaserText,
+                                       scopes: scopes),
+                 client: client,
+                 completion: completion)
+
+            sharedWebCredentialsController = nil
+        }
+
+        sharedWebCredentialsController = SharedWebCredentialsController { [weak self] result in
+            guard case .success((let username, let password)) = result,
+                  let email = EmailAddress(username),
+                  let self = self else {
+                fallback()
+                return
+            }
+
+            self.identityManager.login(username: .email(email),
+                                       password: password,
+                                       scopes: scopes,
+                                       persistUser: true,
+                                       useSharedWebCredentials: false) { result in
+                guard case .success = result else {
+                    fatalError()
```

Comment:
Where is the guarantee login is always succeeding? What if user with stored e-mail is no longer valid or password got changed?

Reply:
This was a typo, meant to be `fallback()`